packages/react-native-sdk/src/components/CallParticipantsList.tsx

```diff
@@ -57,26 +60,33 @@ export const CallParticipantsList = (props: CallParticipantsListProps) => {
   const [_forceUpdateValue, forceUpdate] = useReducer((x) => x + 1, 0);
   const forceUpdateValue = useDebouncedValue(_forceUpdateValue, 500); // we debounce forced value to avoid multiple viewability change continuous rerenders due to callbacks that occurs simultaneously during a large list scroll or when scrolling is completed
 
+  // we use a ref to store the active call object
+  // so that it can be used in the onViewableItemsChanged callback
+  const activeCall = useActiveCall();
+  const activeCallRef = useRef(activeCall);
+  activeCallRef.current = activeCall;
   // This is the function that gets called when the user scrolls the list of participants.
   // It updates viewableParticipantSessionIds HashSet with the session IDs
   // of the participants that are currently visible.
   const onViewableItemsChanged = useRef<
     FlatListProps['onViewableItemsChanged']
-  >(({ viewableItems }) => {
-    // NOTE: viewableItems is more reliable when items are added to the list
-    // but it is not reliable when items are removed from the list.
-    // Hence we clear the HashSet and add the viewable items to it.
-    // We could have also used changed to remove the items from the HashSet one by one, yet we chose to clear the HashSet
-    viewableParticipantSessionIds.current.clear();
-
-    // viewableItems: Visible items are the ones that are currently visible on the screen
-    viewableItems.forEach((viewToken) => {
+  >(({ changed }) => {
+    const participantPatches: StreamVideoParticipantPatches = {};
+    changed.forEach((viewToken) => {
+      participantPatches[viewToken.key] = {
+        viewportVisibilityState: viewToken.isViewable
+          ? VisibilityState.VISIBLE
+          : VisibilityState.INVISIBLE,
+      };
       if (viewToken.isViewable) {
         viewableParticipantSessionIds.current.add(viewToken.key);
+      } else {
+        viewableParticipantSessionIds.current.delete(viewToken.key);
       }
     });
-    if (viewableItems.length) {
+    if (changed.length) {
```

Comment:
Not using `viewableItems` causes participants video not to be rendered when available and fully visible.
This happens when a participant joins a call, see screenshot